examples/openshift-ovs-cni.yml

```diff
@@ -0,0 +1,151 @@
+---
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: ovs-cni-amd64
+  namespace: kube-system
+  labels:
+    tier: node
+    app: ovs-cni
+spec:
+  template:
+    metadata:
+      labels:
+        tier: node
+        app: ovs-cni
+    spec:
+      serviceAccountName: ovs-cni-marker
+      hostNetwork: true
+      nodeSelector:
+        beta.kubernetes.io/arch: amd64
+      tolerations:
+      - key: node-role.kubernetes.io/master
+        operator: Exists
+        effect: NoSchedule
+      containers:
+      - name: ovs-cni-plugin
+        image: quay.io/kubevirt/ovs-cni-plugin:latest
+        imagePullPolicy: IfNotPresent
+        resources:
+          requests:
+            cpu: "100m"
+            memory: "50Mi"
+          limits:
+            cpu: "100m"
+            memory: "50Mi"
+        securityContext:
+          privileged: true
+        volumeMounts:
+        - name: cnibin
+          mountPath: /host/opt/cni/bin
+      - name: ovs-cni-marker
+        image: quay.io/kubevirt/ovs-cni-marker:latest
+        imagePullPolicy: IfNotPresent
+        securityContext:
+          privileged: true
+        args:
+          - -node-name
+          - $(NODE_NAME)
+          - -ovs-socket
+          - unix:///host/run/openvswitch/db.sock
+        volumeMounts:
+          - name: ovs-run
+            mountPath: /host/run/openvswitch
+        env:
+          - name: NODE_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: spec.nodeName
+      volumes:
+        - name: cnibin
+          hostPath:
+            path: /opt/cni/bin
+        - name: ovs-run
+          hostPath:
+            path: /run/openvswitch
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: ovs-cni-marker-cr
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - nodes
+  - nodes/status
+  verbs:
+  - get
+  - update
+  - patch
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: ovs-cni-marker-crb
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: ovs-cni-marker-cr
+subjects:
+- kind: ServiceAccount
+  name: ovs-cni-marker
+  namespace: kube-system
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: ovs-cni-marker
+  namespace: kube-system
+---
+apiVersion: extensions/v1beta1
+kind: DaemonSet
+metadata:
+  name: ovs-vsctl-amd64
+  namespace: kube-system
+  labels:
+    tier: node
+    app: ovs-cni
+spec:
+  template:
+    metadata:
+      labels:
+        tier: node
+        app: ovs-cni
+    spec:
+      nodeSelector:
+        beta.kubernetes.io/arch: amd64
+      tolerations:
+      - key: node-role.kubernetes.io/master
+        operator: Exists
+        effect: NoSchedule
+      containers:
+      - name: ovs-vsctl
+        command:
+        - /bin/bash
+        - -c
+        - |
+          cp /usr/bin/ovs-vsctl /host/usr/bin/_ovs-vsctl
```

Comment:
This is a dark side of ovs-cni.

It has nothing to do with the marker. Our CNI plugin runs completely on the node (unlike openshift-sdn, where CNI binary communicates with a pod through a socket). CNI plugin uses ovs-vsctl availabled on the node. The problem is, openvswitch cannot be installed and running on the host (it breaks openshift-sdn).

So that's why we have this ugly hack. Completely unrelated to this PR. We can do what openshift-sdn does - run openvswitch containerized, keep CNI plugin in a container and only access it through a shim. But that is a big task.

Reply:
Oooh. OK I understand then that's not a problem with this PR, but you probably should create a new issue to track this down.

I understand that it's a huge undertaking. What you perhaps could do in the meantime is not copying this file under /usr/bin but under /opt/cni/bin/ that is already mounted to the container. There are lots of reasons to avoid touching /usr/bin/, from potential breakage of host OVS tools to simple security concerns over mounting this directory - and leaving it with write access at that! - to a container. I think vsctl should work just fine if copied under /opt/cni/bin and not /usr/bin.

Reply:
Aha! I put it under /usr/bin so it is in PATH, but /opt/cni/bin should be added on PATH during CNI binary execution, neat!

Reply:
You can also just call with full path. Anyway, this is not really a discussion for this PR, I don't think so.